src/services/pallets/PalletsForeignAssetsService.ts

```diff
@@ -56,10 +57,23 @@ export class PalletsForeignAssetsService extends AbstractService {
 			if (foreignAssetData) {
 				// remove any commas from multilocation key values e.g. Parachain: 2,125 -> Parachain: 2125
 				const foreignAssetMultiLocationStr = JSON.stringify(foreignAssetData[0]).replace(/(\d),/g, '$1');
-				const foreignAssetMultiLocation = api.registry.createType(
-					'XcmV3MultiLocation',
-					JSON.parse(foreignAssetMultiLocationStr),
-				);
+				const XcmV3MultiLocation = ['StagingXcmV3MultiLocation', 'XcmV3MultiLocation'];
```

Comment:
do we need to create the type from hardcoded options? Bear in mind this will be an issue soon when XCM v4 is life and XCM types are renamed.

Reply:
actually, this should be already the case if foreign assets in Rococo Asset Hub are queried.

Reply:
- You mean retrieve dynamically the `XCM Multilocation` related types, check what is available and try the ones that should work right ? I haven't tried to create types this way but one thing that comes into my mind is the potential impact in the `response time`. 🤔 If there is such an implementation, I am guessing it will take longer to process the request (go through these steps) in order to return the expected result.
- I just tested against Rococo Asset Hub and it returns the foreign assets correctly.

Reply:
- You are right, that solution would be an overkill, but we just need keep an eye for this changes.
- And this makes sense for now, because we are talking about the storage types not the XCM types from the messages. But if Asset hub follows XCM v4 typing, then this will break again, just have it in mind.